trees/trees.go

```diff
@@ -13,6 +13,15 @@ import "github.com/emirpasic/gods/containers"
 
 // Tree interface that all trees implement
 type Tree interface {
+	New() Tree
+	Iterator() containers.ReverseIteratorWithKey
+	Put(key interface{}, value interface{})
+	Remove(key interface{})
+	Get(key interface{}) (interface{}, bool)
+	Keys() []interface{}
+	Min() (interface{}, interface{})
+	Max() (interface{}, interface{})
+
```

Comment:
I see now why you wanted to make the binary heap a container, so the others can implement the interface.

 However, I think that binary heap is still a tree conceptually and should stay that way despite it's internal implementation that has no notion of nodes. 

Also forcing an Iterator() function on all trees makes it a harder job to implement a new tree. These features should be implemented bit-by-bit, a tree, then an iterator, then perhaps an enumerable, etc. but not all at once. Having this strict interface would make contributors life hard and, as seen with the binary heap, not all trees adhere to a key-value logic, some are only values.